### ISSUE TITLE
Harden Rotating Keys support

### ISSUE THREAD

**Issue by ignasi35** [akka/akka#29170]
In https://github.com/akka/akka/pull/29152 a new `SSLEngineProvider` with support for rotating keys was introduced. It caches an `SSLContext` for a certain amount of time. When the cache expires and a new request for an `SSLEngine` comes in creating the new `SSLContext` will read from disk again.

That "read from disk again" operation has a race with any external tool writing the new key set on disk. As a consequence, the `RotatingKeysSSLEngineProvider` introduced in #29152 may fail with `FileNotFoundException` or it could read a partially written file.

The problem is: what should the `RotatingKeysSSLEngineProvider` code do when such a failure happens? Here are some options:

- the node is considered dead and shuts down
- `RotatingKeysSSLEngineProvider` should produce an `SSLContext` with the old keys (kept in memory just in case) and try again at a later time. Include some logs about the issue.
- retry reading the files immediately
- other...

**Comment by patriknw** [comment]
Is that from our tests? In a real environment I would expect that the files are swapped atomically.

**Comment by raboof** [comment]
> Is that from our tests? In a real environment I would expect that the files are swapped atomically.

Yes - and that test was removing the whole directory, that would likely not happen in a real-world scenario. Still, recovering from a failure to rotate the keys might be good to improve.

> RotatingKeysSSLEngineProvider should produce an SSLContext with the old keys (kept in memory just in case) and try again at a later time. Include some logs about the issue.

i think indeed replacing the SSLContext only when it has been loaded successfully would be nicest - but let's also see what fits best.

**Comment by ignasi35** [comment]
@patriknw I mentioned only race conditions but there could be other failure causes:
 - an invalid update on the `cert-manager` setup causing a cascade DoS (e.g. `cert-manager` produces the wrong private key file format)
 - the issuer CA-chain becomes longer than it is supported by `RotatingKeysSSLEngineProvider`

**Comment by ignasi35** [comment]
Another potential improvement could be "pre-rotation validations": running a set of assertions over the newly created `SSLContext` before swapping. That is, even if the file loading succeeds, and the new `SSLContext` can be built it may not be valid for Artery (e.g. a certificate that's only valid for `serverAuth`).

In that case, the new, invalid `SSLContext` is already in the cache and the node can no longer connect to new nodes (only the existing connections are usable).
